src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -709,7 +752,9 @@ await using (var stream = GetInputStream(decompressionStream, snapshotEncryption
             onProgress.Invoke(result.Progress);
 
             var lastFilePath = GetBackupPath(lastFileName);
-            await ImportSingleBackupFile(database, onProgress, result, lastFilePath, context, destination, options, isLastFile: true,
+
+            var des = (subscriptions == null || subscriptions.Count == 0) ? destination : new SnapshotDatabaseDestination(database, subscriptions);
```

Comment:
Pass the `DatabaseDestination` or the `SnapshotDatabaseDestination` to this method

Reply:
Done